### PR TITLE
CIV-10109 Fix ambiguous state error

### PR DESCRIPTION
**Pull request by drummondjm**
https://tools.hmcts.net/jira/browse/CIV-10109

addition to predicate to stop ambiguous state error

modified test to check no ambiguity 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

**Review by sampankumar — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java [hunk 1]
```diff
@@ -397,7 +397,7 @@ public StateFlow build(FlowState.Main initialState) {
                 .transitionTo(CLAIM_DETAILS_NOTIFIED_TIME_EXTENSION)
                     .onlyIf(respondentTimeExtension.and(not(notificationAcknowledged)))
                 //Acknowledging Claim First
-                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged)
+                .transitionTo(NOTIFICATION_ACKNOWLEDGED).onlyIf(notificationAcknowledged.and(not(isInHearingReadiness)))
```
Comment (m-meulendijks-v1) [inline src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:400]: Is this a separate error? Or related to the first?

**Review by m-meulendijks-v1 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java (hunk 1 above)
Comment (m-meulendijks-v1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:400]: I assume it is to cover the DJ journey, or something?

Comment (drummondjm) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:400]: Unrelated, but found during the changes made to test.

          .takenOfflineDate(LocalDateTime.now())
                // ensure no ambiguous transitions between HEARING_READINESS and TAKEN_OFFLINE_AFTER_SDO
                .hearingReferenceNumber("12345")
                .listingOrRelisting(ListingOrRelisting.LISTING)
                .build();

If takenOfflineDate, was null, we would get a ambiguity between NOTIFICATION_ACKNOWLEDGED and HEARING_READINESS.

In practice not a possible scenario afaik, as CCD allowed states in events would not allow it

Comment (m-meulendijks-v1) [reply src/main/java/uk/gov/hmcts/reform/civil/service/flowstate/StateFlowEngine.java:400]: Cool - makes sense :)

**Review by m-meulendijks-v1 — APPROVED** [review]

**Review by vasudevganesanhmcts — APPROVED** [review]

**Pull request merged**